FreeMonoid.agda

```diff
@@ -0,0 +1,72 @@
+module FreeMonoid where
+
+open import Monoid
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality
+import Relation.Binary.SetoidReasoning as ≈-Reasoning
+
+
```

Comment:
Did you know about this trick? When all of your definitions begin with the same parameters, you can create a submodule with those parameters and then you won't have to worry about those parameters while you're inside the module. From outside of the module, you can either provide the parameter when you open the module, or as an extra argument each time you use a function exposed by the module.